version: '3.8'

services:
  db:
    image: mysql:8.0
    container_name: todo_mysql_prod
    environment:
      MYSQL_DATABASE: todo_db
      MYSQL_USER: todo_user
      MYSQL_PASSWORD: todo_password
      MYSQL_ROOT_PASSWORD: root_password
    volumes:
      - mysql_data_prod:/var/lib/mysql
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      timeout: 20s
      retries: 10
    restart: unless-stopped
    networks:
      - production-todo-app-network

  api:
    image: 031988646265.dkr.ecr.us-east-1.amazonaws.com/todo-app:${IMAGE_TAG}
    container_name: todo_django_prod
    expose:
      - "8001"
    volumes:
      - static_volume_prod:/app/staticfiles
    environment:
      DEBUG: "False"
      DB_HOST: db
      DB_NAME: todo_db
      DB_USER: todo_user
      DB_PASSWORD: todo_password
      DB_PORT: 3306
      SECRET_KEY: "django-insecure-3$=+*n%sr4$=+wv32nbh5)-vkzu1bg7f7d%^ndvz5_wz13mis0"
    depends_on:
      db:
        condition: service_healthy
    restart: unless-stopped
    networks:
      - production-todo-app-network

networks:
  production-todo-app-network:
    external: true

volumes:
  mysql_data_prod:
  static_volume_prod:
